src/System.Private.CoreLib/shared/System/Buffer.cs

```diff
@@ -4,6 +4,8 @@
 
 #if AMD64 || ARM64 || (BIT32 && !ARM)
 #define HAS_CUSTOM_BLOCKS
+using Block16 = System.Runtime.Intrinsics.Vector128<byte>;
+using Block32 = System.Runtime.Intrinsics.Vector256<byte>;
```

Comment:
How is this going to behave on ARM64, where `Vector256` is not a TYP_SIMD (is it two 16-byte copies or is it copying 4 ulongs, since that is what `Vector256<T>` has as its fields)?

What about on x86 when AVX is not supported?

Reply:
TYP_SIMD doesn't matter; as there is no field access it will behave the same as all 32 byte structs (like the Block64 previously) and use the largest R2Rable size e.g. the R2R version for x64 is
```asm
; Emitting BLENDED_CODE for X64 CPU with SSE2 - Windows
; ...
G_M6895_IG11:
       movdqu   xmm0, qword ptr [rdx]
       movdqu   qword ptr [rcx], xmm0
       movdqu   xmm0, qword ptr [rdx+16]
       movdqu   qword ptr [rcx+16], xmm0
```
However, because it is a TYP_SIMD the Tier-1 Jit x64 becomes
```asm
; Emitting BLENDED_CODE for X64 CPU with AVX - Windows
; ...
G_M6895_IG11:
       vmovupd  ymm0, ymmword ptr[rdx]
       vmovupd  ymmword ptr[rcx], ymm0
```
Whereas it stays as the 16-byte chunks with a regular struct.

I'm not sure why ARM32 is not included, maybe it faults if not aligned?

Reply:
> TYP_SIMD doesn't matter

I might be mistaken, but I believe that optimization depends on `TYP_SIMD` and therefore `FEATURE_SIMD`.

`FEATURE_SIMD` is not enabled for ARM32, which would explain why it is missing out.

Reply:
Checking how to get arm disasm

Reply:
Not sure how to do ARM; x86 is fine though
```asm
; Emitting BLENDED_CODE for generic X86 CPU - Windows
; ...

G_M6903_IG11:
       movdqu   xmm0, qword ptr [edx]
       movdqu   qword ptr [ecx], xmm0
       movdqu   xmm0, qword ptr [edx+16]
       movdqu   qword ptr [ecx+16], xmm0
```

Reply:
and
```asm
; Emitting BLENDED_CODE for generic X86 CPU - Windows
; Tier-1 compilation

G_M6903_IG11:
       vmovupd  ymm0, ymmword ptr[edx]
       vmovupd  ymmword ptr[ecx], ymm0
```

Reply:
If you have built x86 windows, you should have an arm32 altjit. This will show you what code the jit would produce on arm32.

For jit-diffs you can use this via `--altjit armelnonjit.dll`. 

With corerun you can set `COMPlus_AltJitName=armelnonjit.dll` and `COMPlus_AltJit=*`.

For crossgen, `COMPlus_AltJitName=armelnonjit.dll` and `COMPlus_AltJitNgen=*`.

Reply:
ARM32 using the blocks and without seems to generate the same asm 
```asm
G_M6903_IG11:
            ldr     r1, [r5]
            str     r1, [r4]
            ldr     r1, [r5+4]
            str     r1, [r4+4]
            ldr     r1, [r5+8]
            str     r1, [r4+8]
            ldr     r1, [r5+12]
            str     r1, [r4+12]
            ; ...   etc
```
However; I don't know enough about ARM to take out the non-block path, so will leave it in

Reply:
Without NEON a multi-register instruction seems fastest at +11% (though also uses more registers)
```asm
      LDMIA r1!, {r3 - r10}
      STMIA r0!, {r3 - r10}
```
followed by the asm that is currently generated
```asm
      LDR r3, [r1], #4
      STR r3, [r0], #4
```
http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.faqs/ka13544.html

Reply:
Actually if its the same might as well remove the non-block path as its a lot of code